lib/questionnaires/model/src/questionnaire_item_model.dart

```diff
@@ -185,6 +185,44 @@ class QuestionnaireItemModel extends ChangeNotifier with Diagnosticable {
     }
   }
 
+  /// Returns an integer, starting with 1, that provides the number
+  /// of [QuestionnaireModel]s that have [isAnswerable] flags set to true
+  ///
+  int getQuestionNumber(int answerIndex) {
+    late final int questionNumber;
+
+    /// If [answerIndex] falls within the _cachedAnswerModels data set...
+    /// Check each question in turn until [answerIndex] is reached.
+    /// Create a count of all questions that are labeled as answerable until
+    /// [answerIndexx]
+    ///
+    if (_orderedItems != null) {
+      if (_orderedItems!.length >= answerIndex) {
+        var iterable = 1;
+        for (var idx = 0; idx < answerIndex; idx++) {
+          // Use linked hash map to ensure a key exists at this answer index
+          if (_orderedItems?.keys.elementAt(idx).isNotEmpty ?? false) {
+            // If a key exists, check to see if the isAnswerable flag is true
+            if (_orderedItems?[_orderedItems?.keys.elementAt(idx)]
+                    ?.isAnswerable ??
+                false) {
+              iterable++;
+            }
+          }
+        }
+        questionNumber = iterable;
+      } else {
+        _qimLogger.debug(
```

Comment:
Is this error or debug? I would almost thing this is a programming error when this happens and should throw?

Reply:
It's an error. changes made